nipoppy/workflows/dataset_status.py

```diff
@@ -0,0 +1,225 @@
+        session_ids = doughnut[doughnut.col_session_id].unique()
+
+        self.logger.info(f"Number of participants in doughnut: {len(participant_ids)}")
+        self.logger.info(f"Available visits (n={len(session_ids)}): {session_ids}")
+
+        doughnut_status_df = doughnut.groupby([doughnut.col_session_id]).sum()[
+            [
+                doughnut.col_in_pre_reorg,
+                doughnut.col_in_post_reorg,
+                doughnut.col_in_bids,
+            ]
+        ]
+
+        self.logger.debug(f"doughnut_status_df: {doughnut_status_df}")
+        self.status_df = pd.concat([self.status_df, doughnut_status_df], axis=1)
+
+    def check_bagel(self):
+        """Check the imaging bagel file (if exists)."""
+        nipoppy_checkpoint = "in_imaging_bagel"
+
+        self.logger.info(f"***Status at nipoppy_checkpoint: {nipoppy_checkpoint}***")
+
+        if not self.layout.fpath_imaging_bagel.exists():
+            self.logger.info(
+                f"No bagel file found at {self.layout.fpath_imaging_bagel}."
+            )
+            return
+
+        bagel = Bagel.load(self.layout.fpath_imaging_bagel)
+
+        # Get the number of participants in the doughnut
+        participant_ids = bagel[bagel.col_participant_id].unique()
+        session_ids = bagel[bagel.col_session_id].unique()
+        pipelines = bagel[bagel.col_pipeline_name].unique()
+
+        self.logger.info(f"Number of participants in bagel: {len(participant_ids)}")
+        self.logger.info(f"Available visits (n={len(session_ids)}): {session_ids}")
+        self.logger.info(f"Available pipelines (n={len(pipelines)}): {pipelines}")
+
+        bagel["pipeline"] = (
+            bagel[bagel.col_pipeline_name]
+            + "_"
+            + bagel[bagel.col_pipeline_version]
+            + "_"
+            + bagel[bagel.col_pipeline_step]
+        )
+
+        bagel_pipeline_df = bagel[
+            [
+                bagel.col_participant_id,
+                bagel.col_session_id,
+                "pipeline",
+                bagel.col_status,
+            ]
+        ]
+        bagel_pipeline_df = bagel_pipeline_df[
+            bagel_pipeline_df[bagel.col_status] == STATUS_SUCCESS
+        ]
+
+        bagel_pipeline_df = bagel_pipeline_df.pivot(
+            index=[bagel.col_participant_id, bagel.col_session_id],
+            columns="pipeline",
+            values=bagel.col_status,
+        )
+
+        bagel_status_df = bagel_pipeline_df.groupby([bagel.col_session_id]).count()
+
+        self.logger.debug(f"bagel_status_df: {bagel_status_df}")
+
+        self.status_df = pd.concat([self.status_df, bagel_status_df], axis=1)
+
+    def df_to_table(self):
+        """Convert a pandas.DataFrame obj into a rich.Table obj."""
+        console = Console()
+
+        # Initiate a Table instance
+        title = "Participant counts at each nipoppy checkpoint"
+        table = Table(show_header=True, header_style="bold red", title=title)
```

Comment:
I'm not a huge fan of the big space between the title and the header row, is it possible to reduce it?

Reply:
tired and failed..